File: app/java/app/src/main/java/com/example/accounting/domain/event/JournalDeletedEvent.java
package com.example.accounting.domain.event;

import lombok.Getter;

import java.util.Map;

/**
 * 仕訳削除イベント
 */
@Getter
public class JournalDeletedEvent extends DomainEvent {
    private final String journalNo;
    private final Map<String, Object> oldValues;
    private final String reason;
    private final String userId;
    private final String userName;
    private final String ipAddress;

    public JournalDeletedEvent(
            String journalNo,
            Map<String, Object> oldValues,
            String reason,
            String userId,
            String userName,
            String ipAddress) {
        super();
        this.journalNo = journalNo;
        this.oldValues = oldValues;
        this.reason = reason;
        this.userId = userId;
        this.userName = userName;
        this.ipAddress = ipAddress;
    }

    @Override
    public String getEventType() {
        return "JournalDeleted";
    }
}

File: app/csharp/AccountingSystem.Api/Domain/Events/JournalDeletedEvent.cs
namespace AccountingSystem.Domain.Events;

/// <summary>
/// 仕訳削除イベント
/// </summary>
public record JournalDeletedEvent : DomainEvent
{
    public required string JournalNo { get; init; }
    public required Dictionary<string, object> DeletedData { get; init; }
    public required string UserId { get; init; }
    public required string UserName { get; init; }
    public string? Reason { get; init; }
    public string? IpAddress { get; init; }

    public override string GetEventType() => "JournalDeleted";
}
